Unexpected token ?

Error when installing any of the versions in ubuntu 20.04 LTS, it doesn't matter whether is the old version or the new one:
`../src/unix/pty.cc: In function ‘void pty_after_waitpid(uv_async_t*)’:
../src/unix/pty.cc:512:43: warning: ‘void* memset(void*, int, size_t)’ writing to an object of type ‘class Nan::Persistent<v8::Function>’ with no trivial copy-assignment [-Wclass-memaccess]
  512 |   memset(&baton->cb, -1, sizeof(baton->cb));
      |                                           ^
In file included from ../../nan/nan.h:409,
                 from ../src/unix/pty.cc:20:
../../nan/nan_persistent_12_inl.h:12:40: note: ‘class Nan::Persistent<v8::Function>’ declared here
   12 | template<typename T, typename M> class Persistent :
      |                                        ^~~~~~~~~~
In file included from ../../nan/nan.h:60,
                 from ../src/unix/pty.cc:20:
../src/unix/pty.cc: At global scope:
/usr/include/nodejs/src/node.h:573:43: warning: cast between incompatible function types from ‘void (*)(Nan::ADDON_REGISTER_FUNCTION_ARGS_TYPE)’ {aka ‘void (*)(v8::Local<v8::Object>)’} to ‘node::addon_register_func’ {aka ‘void (*)(v8::Local<v8::Object>, v8::Local<v8::Value>, void*)’} [-Wcast-function-type]
  573 |       (node::addon_register_func) (regfunc),                          \
      |                                           ^
/usr/include/nodejs/src/node.h:607:3: note: in expansion of macro ‘NODE_MODULE_X’
  607 |   NODE_MODULE_X(modname, regfunc, NULL, 0)  // NOLINT (readability/null_usage)
      |   ^~~~~~~~~~~~~
../src/unix/pty.cc:734:1: note: in expansion of macro ‘NODE_MODULE’
  734 | NODE_MODULE(pty, init)
      | ^~~~~~~~~~~
In file included from /usr/include/nodejs/src/node.h:63,
                 from ../../nan/nan.h:60,
                 from ../src/unix/pty.cc:20:
/usr/include/nodejs/deps/v8/include/v8.h: In instantiation of ‘void v8::PersistentBase<T>::SetWeak(P*, typename v8::WeakCallbackInfo<P>::Callback, v8::WeakCallbackType) [with P = node::ObjectWrap; T = v8::Object; typename v8::WeakCallbackInfo<P>::Callback = void (*)(const v8::WeakCallbackInfo<node::ObjectWrap>&)]’:
/usr/include/nodejs/src/node_object_wrap.h:84:78:   required from here
/usr/include/nodejs/deps/v8/include/v8.h:9502:16: warning: cast between incompatible function types from ‘v8::WeakCallbackInfo<node::ObjectWrap>::Callback’ {aka ‘void (*)(const v8::WeakCallbackInfo<node::ObjectWrap>&)’} to ‘Callback’ {aka ‘void (*)(const v8::WeakCallbackInfo<void>&)’} [-Wcast-function-type]
 9502 |                reinterpret_cast<Callback>(callback), type);
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/nodejs/deps/v8/include/v8.h: In instantiation of ‘void v8::PersistentBase<T>::SetWeak(P*, typename v8::WeakCallbackInfo<P>::Callback, v8::WeakCallbackType) [with P = Nan::ObjectWrap; T = v8::Object; typename v8::WeakCallbackInfo<P>::Callback = void (*)(const v8::WeakCallbackInfo<Nan::ObjectWrap>&)]’:
../../nan/nan_object_wrap.h:65:61:   required from here
/usr/include/nodejs/deps/v8/include/v8.h:9502:16: warning: cast between incompatible function types from ‘v8::WeakCallbackInfo<Nan::ObjectWrap>::Callback’ {aka ‘void (*)(const v8::WeakCallbackInfo<Nan::ObjectWrap>&)’} to ‘Callback’ {aka ‘void (*)(const v8::WeakCallbackInfo<void>&)’} [-Wcast-function-type]
Unexpected token ?
`

--- a/index.js
+++ b/index.js
@@ -7,6 +7,7 @@ const { Server } = require("socket.io");
 const { io } = require("socket.io-client");
 const term = require( 'terminal-kit' ).terminal;
 const Downloader = require("nodejs-file-downloader");
+const semver = require('semver');
 const platform = os.platform()
 const shell = platform === 'win32' ? 'powershell.exe' : 'bash';
 class Dalai {
@@ -117,6 +118,12 @@ class Dalai {
     return modelNames
   }
   async install(...models) {
+    // Check if current version is greater than or equal to 18
+    const node_version = process.version;
+    if (!semver.gte(node_version, '18.0.0')) {
+      throw new Error("outdated Node version, please install Node 18 or newer")
+    }
+
     // install llama.cpp to home
     let success = await this.exec(`git clone https://github.com/ggerganov/llama.cpp.git ${this.home}`)
     if (!success) {

--- a/package.json
+++ b/package.json
@@ -12,13 +12,14 @@
     "just:fix": "npx prettier --write ."
   },
   "dependencies": {
+    "ejs": "^3.1.8",
+    "express": "^4.18.2",
     "node-pty": "^0.10.1",
     "nodejs-file-downloader": "^4.10.6",
+    "semver": "^7.3.8",
     "socket.io": "^4.6.1",
     "socket.io-client": "^4.6.1",
-    "terminal-kit": "^3.0.0",
-    "ejs": "^3.1.8",
-    "express": "^4.18.2"
+    "terminal-kit": "^3.0.0"
   },
   "devDependencies": {
     "prettier": "^2.8.4"

